pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -321,20 +323,20 @@ func needPodCIDRsUpdate(node *v1.Node, podCIDRs []*net.IPNet) (bool, error) {
 	}
 	_, nodePodCIDR, err := netutils.ParseCIDRSloppy(node.Spec.PodCIDR)
 	if err != nil {
-		klog.ErrorS(err, "Found invalid node.Spec.PodCIDR", "node.Spec.PodCIDR", node.Spec.PodCIDR)
+		klog.ErrorS(err, "Found invalid node.Spec.PodCIDR", "podCIDR", node.Spec.PodCIDR)
 		// We will try to overwrite with new CIDR(s)
 		return true, nil
 	}
 	nodePodCIDRs, err := netutils.ParseCIDRs(node.Spec.PodCIDRs)
 	if err != nil {
-		klog.ErrorS(err, "Found invalid node.Spec.PodCIDRs", "node.Spec.PodCIDRs", node.Spec.PodCIDRs)
+		klog.ErrorS(err, "Found invalid node.Spec.PodCIDRs", "podCIDRs", node.Spec.PodCIDRs)
 		// We will try to overwrite with new CIDR(s)
 		return true, nil
 	}
 
 	if len(podCIDRs) == 1 {
 		if cmp.Equal(nodePodCIDR, podCIDRs[0]) {
-			klog.V(4).InfoS("Node already has allocated CIDR. It matches the proposed one.", "nodeName", node.Name, "podCIDRs[0]", podCIDRs[0])
+			klog.V(4).InfoS("Node already has allocated CIDR. It matches the proposed one", "nodeName", node.Name, "podCIDRs[0]", podCIDRs[0])
```

Comment:
`podCIRDs[0]` is not a valid key according to the guidelines. Perhaps just `podCIDR`?

The logcheck tool only checks for ASCII characters, which is much too forgiving.

Reply:
Yeah, You're right. key contains `spaces` or `[]`  is not a valid key, which is always forgotten sometimes.
I'm interested in looking into whether the logcheck tool can do the check